pyquil/reference_simulator.py

```diff
@@ -246,8 +267,20 @@ def expectation(self, operator: Union[PauliTerm, PauliSum]):
         raise NotImplementedError("To implement")
 
     def reset(self) -> 'AbstractQuantumSimulator':
-        self.density.fill(0)
-        self.density[0, 0] = complex(1.0, 0)
+        """
+        Resets the initial state of ReferenceDensitySimulator.
+
+        If ``self.initial_density`` is ``None`` the QVM state, that is ``self.density``,
+        is reset to the all zeros state. Otherwise ``self.density`` is reset to the user specified
+        ``self.initial_density``.
+
+        :return: ``self`` to support method chaining.
+        """
+        if self.initial_density is None:
```

Comment:
Small nit to pick: you can reduce this to just `if self.initial_density: ...` by swapping the bodies of the if/else. I have no strong preference other than style.

Reply:
- [x] todo

Reply:
Weird. NVM then :)